### PR TITLE
Fix Master.JournalSequenceNumber metrics in RaftJournalSystem

### PR DESCRIPTION
**Pull request by kaijchen**
### What changes are proposed in this pull request?

Fix Master.JournalSequenceNumber metrics and add RaftJournalSystemMetricsTest.

### Why are the changes needed?

Fix Master.JournalSequenceNumber metrics.
Test Metrics registered in `RaftJournalSystem#initServer()`.

### Does this PR introduce any user facing changes?

No.

### CODE REVIEW

**Comment by kaijchen** [comment]
Looks like there are some issues in when the `RaftJournalSystem` calls `AbstractJournalSystem#registerMetrics()`.

**Comment by kaijchen** [comment]
Cc @maobaolong @jiacheliu3 PTAL.

**Review by elega — APPROVED** [review]
LGTM. Thanks for the thorough testing

File: core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java [hunk 1]
```diff
@@ -545,6 +545,7 @@ public synchronized void gainPrimacy() {
     mAsyncJournalWriter
         .set(new AsyncJournalWriter(mRaftJournalWriter, () -> getJournalSinks(null)));
     mTransferLeaderAllowed.set(true);
+    super.registerMetrics();
```
Comment (kaijchen) [inline core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java:548]: I'm not sure whether this change is required now.

File: core/server/master/src/test/java/alluxio/master/journal/raft/RaftJournalSystemMetricsTest.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        .setFollowerInfo(RaftProtos.FollowerInfoProto.newBuilder()
+            .setLeaderInfo(RaftProtos.ServerRpcProto.newBuilder()
+                .setId(RaftProtos.RaftPeerProto.newBuilder()
+                    .setId(ByteString.copyFromUtf8("localhost_19201")))))
+        .build();
+
+    Map<String, Long> sn1 = new HashMap<String, Long>() {
+      {
+        put("foo", 1L);
+      }
```
Comment (jiacheliu3) [inline core/server/master/src/test/java/alluxio/master/journal/raft/RaftJournalSystemMetricsTest.java:68]: nit: you can use `ImmutableMap.of()`

**Review by jiacheliu3 — APPROVED** [review]
LGTM with a minor comment. Thanks for the fix!

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**